Add getters and setters for height and color options

tf opened the pull request:
Allow updating the `waveColor`, `progressColor`, `cursorColor` and `height` options that were passed as part of the options for create.

For the wave color options, simply ensure that the wave is redrawn. For the height option, delegate to `drawer.setHeight` to handle the update.  Note that - while drawer applies the `pixelRatio` itself when initializing from `params` - `setHeight` expects a height that already has been multiplied by `pixelRatio`.

The cursor is a border on one of the wave elements. Add an `updateCursor` method to the drawer interface to allow updating the cursor related styles.

Redrawing the wave via `drawBuffer` only works correctly after #1144 has been merged.

agamemnus commented:
`it('should allow setting waveColor', function () {`...

tf commented:
@agamemnus what does that mean?

agamemnus commented:
That's in your code...

On Jun 20, 2017 8:21 AM, "Tim Fischbach" <notifications@github.com> wrote:

> @agamemnus <https://github.com/agamemnus> what does that mean?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/katspaugh/wavesurfer.js/pull/1145#issuecomment-309736107>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADFJWNkypE37papVhc4pEVT4fWdzJxiZks5sF7k3gaJpZM4N_aKG>
> .
>

tf commented:
I know. What do you want me to change?

agamemnus commented:
I don't see it working unless I'm missing something. Seems like you're
using a library or something.

On Jun 20, 2017 8:57 AM, "Tim Fischbach" <notifications@github.com> wrote:

> I know. What do you want me to change?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/katspaugh/wavesurfer.js/pull/1145#issuecomment-309744607>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADFJWKinFsj5OGkjIRCabi1VSA9J6NpJks5sF8GtgaJpZM4N_aKG>
> .
>

tf commented:
Still not quite sure we are talking about the same thing, but there is a test suite that already covers things like [toggling mute](https://github.com/katspaugh/wavesurfer.js/blob/master/spec/wavesurfer.spec.js#L65), so I also added some tests for the getters and setters that I added.

> I don't see it working

The tests run [on Travis](https://travis-ci.org/katspaugh/wavesurfer.js/builds/244886876#L376). The new ones are also passing.

agamemnus commented:
Oh, ok (I was on my phone).

tf commented:
I'll also add getters/setters for `cursorColor` to cover all color options. I'll update the PR tomorrow.

agamemnus commented:
I have a different method that I put as a PR. Basically using get and set:

https://github.com/katspaugh/wavesurfer.js/pull/1132
https://github.com/katspaugh/wavesurfer.js/pull/1130

tf commented:
I've updated the PR to also allow updating the cursor color.

@agamemnus From my point of view, the methods added by this PR fit in quite nicely with the existing API and completely cover my use case. So I'll leave this PR like this, as one option. If things are supposed to move in a different direction, though, I'm happy to port my application to a different API once a future release of Wavesurfer provides the features added here.

tf commented:
Sure. Just I'll update the PR this week.

tf commented:
@thijstriemstra Added the inline docs. Tried to follow the conventions found on the `next` branch.

tf commented:
@thijstriemstra All of a sudden I am seeing a whole bunch of unrelated commits in this and my two other PRs. Shall I recreate the PRs? What happened?

mspae commented:
@tf the next (v2) branch was set to be the master branch, and the old master is now es5 (version 1) – you need to rebase your commits onto the new master branch.

katspaugh commented:
@tf you can also change the base branch from this PR page directly.

tf commented:
@katspaugh @thijstriemstra rebased and updated.
